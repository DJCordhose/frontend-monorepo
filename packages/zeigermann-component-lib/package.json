{
  "name": "zeigermann-component-lib",
  "version": "0.1.17",
  "license": "MIT",
  "main": "dist/index.cjs.js",
  "module": "dist/index.esm.js",
  "types": "dist/index.d.ts",
  "source": "src/index.ts",
  "files": [
    "/dist",
    "/src",
    "LICENSE.md"
  ],
  "prepublish": "tsc",
  "scripts": {
    "build": "rollup -c",
    "build:watch": "rollup -c --watch",
    "build:ts": "tsc --build",
    "build:ts:watch": "tsc --build --watch",
    "prepublish": "yarn build",
    "clean": "rm -rf dist",
    "test": "jest",
    "test:watch": "jest --watch",
    "storybook": "start-storybook -s ./public -p 6006",
    "build-storybook": "build-storybook"
  },
  "gitHead": "7c665356e45d81678fce7f080d46aea688f52d7a",
  "devDependencies": {
    "@babel/core": "^7.14.3",
    "@rollup/plugin-commonjs": "^19.0.0",
    "@rollup/plugin-node-resolve": "^13.0.0",
    "@rollup/plugin-typescript": "^8.2.1",
    "@storybook/addon-actions": "^6.2.9",
    "@storybook/addon-essentials": "^6.2.9",
    "@storybook/addon-links": "^6.2.9",
    "@storybook/react": "^6.2.9",
    "@testing-library/jest-dom": "^5.12.0",
    "@testing-library/react": "^11.2.7",
    "@types/jest": "^26.0.23",
    "@types/react": "^17.0.6",
    "babel-loader": "^8.2.2",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^27.0.3",
    "postcss": "^8.3.0",
    "rollup": "^2.50.5",
    "rollup-plugin-delete": "^2.0.0",
    "rollup-plugin-peer-deps-external": "^2.2.4",
    "rollup-plugin-postcss": "^4.0.0",
    "rollup-plugin-postcss-modules": "^2.0.2",
    "rollup-plugin-terser": "^7.0.2",
    "rollup-plugin-typescript2": "^0.30.0",
    "storybook-css-modules-preset": "^1.1.1",
    "ts-jest": "^27.0.2",
    "ts-node": "^10.0.0",
    "typescript": "^4.3.2"
  },
  "peerDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "dependencies": {
    "@testing-library/user-event": "^13.1.9"
  }
}
