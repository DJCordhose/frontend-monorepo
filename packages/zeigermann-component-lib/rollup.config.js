// import typescript from "@rollup/plugin-typescript";
// creates d.ts files for everthing
import typescript from 'rollup-plugin-typescript2';
import external from "rollup-plugin-peer-deps-external";
import resolve from "@rollup/plugin-node-resolve";
import commonjs from "@rollup/plugin-commonjs";
// we probably want full sources
import { terser } from 'rollup-plugin-terser';
// import postcss from "rollup-plugin-postcss";
// can create nice css.d.ts files
import postcss from 'rollup-plugin-postcss-modules';
import del from "rollup-plugin-delete";
import pkg from "./package.json";

export default {
  // preserveModules: true,
  input: pkg.source,
  output: [
    { file: pkg.main, format: "cjs", sourcemap: true },
    { file: pkg.module, format: "esm", sourcemap: true },
  ],
  plugins: [
    postcss({
      writeDefinitions: true,
      modules: true,
      sourceMap: true,
      extract: true,
      minimize: true,
    }),
    external(),
    resolve(),
    commonjs(),
    typescript(),
    del({ targets: ["dist/*"] }),
    // terser()
  ],
  external: Object.keys(pkg.peerDependencies || {}),
};
