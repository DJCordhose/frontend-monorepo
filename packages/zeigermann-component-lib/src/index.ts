export * from "./AriaButton"
export * from "./FlexContainer"
export * from "./LoadingIndicator"
export * from "./util"