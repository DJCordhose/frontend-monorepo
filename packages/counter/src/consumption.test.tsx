import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import { render, screen } from "@testing-library/react";

import { AriaButton, FlexContainer } from "zeigermann-component-lib";
test("AriaButton renders as expected", () => {
  const onClick = jest.fn();

  const tree = renderer
    .create(
      <AriaButton
        testid="language:switch:de"
        label="Change Language"
        text="Deutsch"
        onClick={onClick}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test("AriaButton click triggers callback as expected", () => {
  const onClick = jest.fn();
  const label = "Change Language";

  render(
    <AriaButton
      testid="language:switch:de"
      label={label}
      text="Deutsch"
      onClick={onClick}
    />
  );

  userEvent.click(screen.getByLabelText(label));
  expect(onClick).toHaveBeenCalled();
});

test("FlexContainer renders as expected", () => {
  const tree = renderer
    .create(
      <>
        <FlexContainer>
          <p></p>
        </FlexContainer>
        <FlexContainer center={true}>
          <p></p>
        </FlexContainer>
      </>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
