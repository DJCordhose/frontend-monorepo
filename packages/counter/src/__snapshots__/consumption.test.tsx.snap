// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AriaButton renders as expected 1`] = `
<button
  aria-label="Change Language"
  className="
      AriaButton-module_button__2nCm9
      bg-gray-100 border-2 border-gray-900
      hover:border-gray-500 hover:bg-gray-200 
      active:border-gray-200 active:bg-gray-400 
      px-2 py-2 w-40 mx-2 my-2 "
  data-testid="language:switch:de"
  onClick={[MockFunction]}
>
  Deutsch
</button>
`;

exports[`FlexContainer renders as expected 1`] = `
Array [
  <div
    className="
      flex justify-between border-gray-900 border-2 m-2 overflow-hidden"
  >
    <p />
  </div>,
  <div
    className="
      flex justify-center border-gray-900 border-2 m-2 overflow-hidden"
  >
    <p />
  </div>,
]
`;
